miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSequencerReferenceController.java

```diff
@@ -25,111 +25,91 @@
-    if (user.isAdmin()) {
-      return new ModelAndView("redirect:/miso/stats/sequencer/" + referenceId);
-    }
-    else {
-      SequencerReference sr = sequencerReferenceService.get(referenceId);
-      Collection<SequencerServiceRecord> serviceRecords = sequencerReferenceService.listServiceRecordsByInstrument(referenceId);
-      
-      if (sr != null) {
-        model.put(ModelKeys.SEQUENCER.getKey(), sr);
-        model.put(ModelKeys.RECORDS.getKey(), serviceRecords);
-        model.put("title", "Edit Sequencer");
-        String ip = sr.getIpAddress() == null ? "" : sr.getIpAddress().toString();
-        if (ip.startsWith("/")) {
-          model.put(ModelKeys.TRIMMED_IP.getKey(), ip.substring(1));
-        } else {
-          model.put(ModelKeys.TRIMMED_IP.getKey(), ip);
-        }
-      } else {
-        throw new IOException("Cannot retrieve the requested sequencer reference");
-      }
-      return new ModelAndView("/pages/editSequencerReference.jsp", model);
+    SequencerReference sr = sequencerReferenceService.get(referenceId);
+    Collection<SequencerServiceRecord> serviceRecords = sequencerReferenceService.listServiceRecordsByInstrument(referenceId);
+
+    if (sr == null) {
+      throw new IOException("Cannot retrieve the requested sequencer reference");
     }
-  }
-  
-  @RequestMapping("/servicerecord/{recordId}")
-  public ModelAndView viewServiceRecord(@PathVariable("recordId") Long recordId, ModelMap model) throws IOException {
-    User user = securityManager.getUserByLoginName(SecurityContextHolder.getContext().getAuthentication().getName());
     if (user.isAdmin()) {
-      return new ModelAndView("redirect:/miso/stats/sequencer/servicerecord/" + recordId);
+      model.put("otherSequencerReferences",
+          sequencerReferenceService.list().stream().filter(other -> other.getId() != referenceId).collect(Collectors.toList()));
     }
-    else {
-      return editServiceRecordController.viewServiceRecord(recordId, model);
+    model.put("preUpgradeSeqRef", sequencerReferenceService.getByUpgradedReferenceId(sr.getId()));
+
+    model.put("sequencerReference", sr);
+    model.put("sequencerServiceRecords", serviceRecords);
+    model.put("title", "Edit Sequencer");
+    String ip = sr.getIpAddress() == null ? "" : sr.getIpAddress().toString();
+    model.put("trimmedIpAddress", ip.startsWith("/") ? ip.substring(1) : ip);
+    return new ModelAndView("/pages/editSequencerReference.jsp", model);
+  }
+
+  @RequestMapping(method = RequestMethod.POST)
+  public String processSubmit(@ModelAttribute("sequencerReference") SequencerReference sr, ModelMap model, SessionStatus session)
+      throws IOException {
+    Long srId = null;
+    if (sr.getId() == SequencerReferenceImpl.UNSAVED_ID) {
+      srId = sequencerReferenceService.create(sr);
+    } else {
+      sequencerReferenceService.update(sr);
+      srId = sr.getId();
     }
+    return "redirect:/miso/sequencer/" + srId;
```

Comment:
Editing a sequencer saves successfully, but sets the page's URL to this: `http://localhost:8080/miso/sequencer/49?misoBugUrl=https%3A%2F%2Fjira.oicr.on.ca%2Fsecure%2FCreateIssueDetails%21init.jspa%3Fissuetype%3D1%26pid%3D11685&misoInstanceName=Heather%27s+Test+Kitchen`